test/parametrized_tester.c

```diff
@@ -0,0 +1,194 @@
+/* parametrized_tester.c - Test deflate() and inflate() on a chunk of the
+ * input data file, combining many deflate() settings to improve testing coverage.
+ */
+#include "zbuild.h"
+#ifdef ZLIB_COMPAT
+#  include "zlib.h"
+#else
+#  include "zlib-ng.h"
+#endif
+
+#include <stdio.h>
+
+/*
+ * WITH_EXTRA_TESTS results in 9450 permutations per input file.
+ * Normal mode results in 540.
+ *
+ * WITH_EXTRA_TESTS also tests a bigger chunk of the files.
+ */
+
+#ifdef WITH_EXTRA_TESTS
+const int valid_wbits[] = {
+    -9, -10, -11, -12, -13, -14, -15,
+     9,  10,  11,  12,  13,  14,  15,
+# ifdef WITH_GZFILEOP
+    25,  26,  27,  28,  29,  30,  31,
+# endif
+};
+
+const int valid_compression_levels[] = {
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9
+};
+
+const int valid_mem_levels[] = {
+    1, 2, 3, 4, 5, 6, 7, 8, 9,
+};
+
+#define MAX_FILESIZE_KB 4096
+
+# ifdef WITH_GZFILEOP
+#  define NUM_W 21
```

Comment:
NUM_W? Consider a better name? Same with NUM_C, NUM_M.

Reply:
I did not like these names, but I couldn't come up with any that I liked. Ideas?